Fix test failure on ubuntu-0.10

bajtos opened the pull request:
- Rework the tests checking for presence of resource elements to activelly poll until the element is present. This should reduce likelihood of a timing issue where a simple assertion on element presence
  could be called before the swagger JSON is parsed and the UI
  constructed.
- Improve browser logging to reliably report `console.log` errors
- Fix CI detection to detect Jenkins as well as Travis
- Disable browser tests on ubuntu-0.10 as it's unclear why they are failing on this (and only on this) slave

Commit:
test: wait for swagger resource to be loaded

Rework the tests checking for presence of resource elements to
activelly poll until the element is present.

This fixes a timing issue where a simple assertion on element presence
could be called before the swagger JSON is parsed and the UI
constructed.

rmg commented on test/e2e/servers.js:
```diff
@@ -29,7 +29,7 @@ module.exports.start = function (specsLocation, done) {
     driver.get(url);
     done();
-  }, process.env.TRAVIS ? 20000 : 5000);
+  }, (process.env.TRAVIS || process.env.JENKINS_HOME) ? 20000 : 5000);
```
Both of these CI environments also set `process.env.CI` :-)

rmg commented:
CI is also running coverage, which can dramatically slow down tests.

bajtos commented:
@rmg It looks like Firefox on ubuntu-0.10 does not like the client javascript code, even though the same code runs fine on all other CI slaves.

Error message:

```
browser error message: TypeError: setting a property that has only a getter
browser error message: TypeError: SwaggerUi.Views.HeaderView is not a constructor
```

To make the issue even more confusing, the browser version seems to be the same as on other build slaves: `BROWSER NAME firefox VERSION 40.0.3`

Commit:
Improve browser logging

  - Fix detection of console.error logs
  - Log non-error items too if DEBUG is truthy
  - Log browser name and version if DEBUG is truthy

Commit:
Fix CI detection

Use the env variable `CI` instead of `TRAVIS`

Commit:
Disable browser tests on ubuntu-0.10

The tests are always failing on this slave, while they are reliably
passing on all other Jenkins slaves.

bajtos commented:
After spending a good day trying to figure out what's wrong, I am giving up.

I have added a condition to disable the tests on ubuntu-0.10.

If we think we should invest more time into investigating the problem, then I will create a follow-up user story to be planned in a future sprint.

@rmg could you please review code changes?

/cc @chandadharap

bajtos commented:
FWIW, the main reason for giving up is that I could not find a way how to get the location where `TypeError: setting a property that has only a getter` was thrown. Selenium WebDriver does not forward location data in the console log entries, and `window.onerror` is ignored when the page is opened via webdriver for Firefox.

rmg commented:
It is possible that the Ubuntu/v0.10 image older than the others.. they get a little out of sync when only one major version of node gets a minor/patch release.

rmg commented:
@slnode test please

rmg commented:
Sorry for the CI spam, just wanted to check if the updated CI images fixed the problem. They did not.

As for the oddness about only that one configuration misbehaving.. Perhaps it is a regression in the Selenium wrappers from when node-v0.12/v4 support was added?

LGTM :-)
